Woof/Woof/ViewModel/SitterProfileViewModel.swift

```diff
@@ -91,35 +84,55 @@ final class SitterProfileViewModel: ObservableObject {
 
     private func upload(_ sitter: Sitter) async throws {
         let endpoint = WoofAppEndpoint.addNewSitter(sitter.asDictionary())
-        _ = try await NetworkService<WoofAppEndpoint>().request(endpoint)
+
+        do {
+            _ = try await NetworkService<WoofAppEndpoint>().request(endpoint)
+        } catch {
+            throw AppError.uploadFailed
+        }
     }
 
-    private func saveLocally(_ sitter: Sitter) async {
+    private func saveLocally(_ sitter: Sitter) async throws {
         guard let data = try? JSONEncoder().encode(sitter),
               KeyValueStorage(KeyValueStorage.Name.currentSitter)
               .save(data, for: KeyValueStorage.Key.currentSitter) else {
-            await handleError(.saveLocallyFailed)
+            throw AppError.saveLocallyFailed
+        }
+    }
+
+    @MainActor private func handleError(_ error: Error) {
+        guard let appError = error as? AppError else {
+            errorMessage = "An error occurred."
             return
         }
+
+        errorMessage = appError.errorDescription
     }
 
-    @MainActor private func handleError(_ error: AppError) {
-        errorMessage = error.errorDescription
+    /// Indicates whether there is a saved sitter.
+    private(set) var sitterIsSet = false
```

Comment:
Is it private property?

Reply:
No, you have access to this property throughout the app.
Fixed with e9d20b2